server/src/main/java/org/opensearch/action/search/TransportDeletePitAction.java

```diff
@@ -0,0 +1,101 @@
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService,
+        PitService pitService
+    ) {
+        super(DeletePitAction.NAME, transportService, actionFilters, DeletePitRequest::new);
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+        this.pitService = pitService;
+    }
+
+    /**
+     * Invoke 'delete all pits' or 'delete list of pits' workflow based on request
+     */
+    @Override
+    protected void doExecute(Task task, DeletePitRequest request, ActionListener<DeletePitResponse> listener) {
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+            deleteAllPits(listener);
+        } else {
+            deletePits(listener, request);
+        }
+    }
+
+    /**
+     * Deletes one or more point in time search contexts.
+     */
+    private void deletePits(ActionListener<DeletePitResponse> listener, DeletePitRequest request) {
+        Map<String, List<PitSearchContextIdForNode>> nodeToContextsMap = new HashMap<>();
+        for (String pitId : request.getPitIds()) {
+            SearchContextId contextId = SearchContextId.decode(namedWriteableRegistry, pitId);
+            for (SearchContextIdForNode contextIdForNode : contextId.shards().values()) {
+                PitSearchContextIdForNode pitSearchContext = new PitSearchContextIdForNode(pitId, contextIdForNode);
+                List<PitSearchContextIdForNode> contexts = nodeToContextsMap.getOrDefault(contextIdForNode.getNode(), new ArrayList<>());
+                contexts.add(pitSearchContext);
+                nodeToContextsMap.put(contextIdForNode.getNode(), contexts);
+            }
+        }
+        pitService.deletePitContexts(nodeToContextsMap, listener);
+    }
+
+    /**
+     * Delete all active PIT reader contexts
+     */
+    private void deleteAllPits(ActionListener<DeletePitResponse> listener) {
+        // TODO: Use list all PITs to delete all PITs in case of remote cluster use case
+        int size = clusterService.state().getNodes().getSize();
+        ActionListener groupedActionListener = pitService.getDeletePitGroupedListener(listener, size);
+        for (final DiscoveryNode node : clusterService.state().getNodes()) {
+            try {
+                Transport.Connection connection = searchTransportService.getConnection(null, node);
+                searchTransportService.sendFreeAllPitContexts(connection, groupedActionListener);
+            } catch (Exception e) {
+                groupedActionListener.onFailure(e);
+            }
```

Comment:
What happens to a PIT spanning remote cluster

Reply:
Will be handled as part of Get all pits changes - https://github.com/bharath-techie/OpenSearch/pull/5/files#diff-ffb8766908179bfde8269b46038d1e17d4997da9214f5adc66a4bd08d5ba1727R90